cimgen/languages/modernpython/lang_pack.py

```diff
@@ -76,23 +79,110 @@ def _get_type_and_default(text, render) -> tuple[str, str]:
+            and attribute["about"]
+            and "multiplier" in attribute["about"]
+            and "isFixed" in attribute
+        ):
+            datatype_attributes["multiplier"] = "UnitMultiplier." + attribute["isFixed"]
+        if "about" in attribute and attribute["about"] and "unit" in attribute["about"] and "isFixed" in attribute:
+            datatype_attributes["unit"] = "UnitSymbol." + attribute["isFixed"]
+    return datatype_attributes
+
+
+def _set_attribute_class(text, render) -> str:
+    attribute = eval(render(text))
+    attribute_class = ""
+    if attribute["is_datatype_attribute"] or attribute["is_primitive_attribute"]:
+        attribute_class += "\n            "
+        attribute_class += """"attribute_class": """ + attribute["attribute_class"] + ","
+    return attribute_class
+
+
 def run_template(output_path, class_details):
-    if class_details["is_a_primitive_class"] or class_details["is_a_datatype_class"]:
-        return
-    for template_info in template_files:
-        resource_file = Path(
-            os.path.join(
-                output_path,
-                "resources",
-                class_details["class_name"] + template_info["ext"],
-            )
-        )
-        if not resource_file.exists():
-            if not (parent := resource_file.parent).exists():
-                parent.mkdir()
+    if class_details["is_a_primitive_class"]:
+        # Primitives are never used in the in memory representation but only for
+        # the schema
+        template = primitive_template_file
+        class_details["python_type"] = _primitive_to_data_type(class_details["class_name"])
+    elif class_details["is_a_datatype_class"]:
+        # Datatypes based on primitives are never used in the in memory
+        # representation but only for the schema
+        template = datatype_template_file
+        class_details.update(_set_datatype_attributes(class_details["attributes"]))
+    elif class_details["is_an_enum_class"]:
+        template = enum_template_file
+    else:
+        template = class_template_file
         class_details["setDefault"] = _set_default
         class_details["setType"] = _set_type
-        _write_templated_file(resource_file, class_details, template_info["filename"])
+        class_details["setAttributeClass"] = _set_attribute_class
+        class_details["setImports"] = _set_imports(class_details["attributes"])
+    resource_file = _create_file(output_path, class_details, template)
+    _write_templated_file(resource_file, class_details, template["filename"])
+
+
+def _create_file(output_path, class_details, template) -> str:
+    resource_file = Path(
+        os.path.join(
+            output_path,
+            "resources",
+            class_details["class_name"] + template["ext"],
+        )
+    )
+    if not resource_file.exists():
+        if not (parent := resource_file.parent).exists():
+            parent.mkdir()
+    return resource_file
```

Comment:
```suggestion
    return str(resource_file)
```
I didn't try. But this should solve the sonarcloud issue.